backend/YouTubeV2.Application/Services/VideoServices/VideoService.cs

```diff
@@ -48,5 +48,27 @@ public async Task SetVideoProcessingProgressAsync(Video video, ProcessingProgres
             video.ProcessingProgress = processingProgress;
             await _context.SaveChangesAsync(cancellationToken);
         }
+
+        public async Task<VideoMetadataDto> GetVideoMetadataAsync(Guid id, CancellationToken cancellationToken = default)
+        {
+            var video = await _context.Videos.Include(video => video.Tags).SingleAsync(video => video.Id == id, cancellationToken);
+            var user = await _context.Users.SingleAsync(user => user.Videos.Contains(video), cancellationToken);
+            var thumbnail = _blobImageService.GetVideoThumbnail(video.Id.ToString());
+
+            return new VideoMetadataDto(
+                video.Id.ToString(), 
+                video.Title, 
+                video.Description,
+                thumbnail.ToString(),
+                user.Id,
+                user.UserName,
+                video.ViewCount,
+                video.Tags.Select(tag => tag.ToString()).ToList(),
+                video.Visibility.ToString(),
+                video.ProcessingProgress.ToString(),
+                video.UploadDate.DateTime,
+                video.EditDate.DateTime,
+                video.Duration);
```

Comment:
where do we set the duration of the video? I think we just dont maybe it requires a fix